baqend/_resultGeneration.ts

```diff
@@ -77,9 +77,50 @@ export async function generateTestResult(wptTestId: string, pendingTest: model.T
   // Now the test is finished with data
   pendingTest.testDataMissing = false
 
+  const wptWaterfall = await createWaterfall(wptTestId, url, isDesktop, db);
+  const wptFilmstrip = await createFilmStrip(wptTestId, url, isDesktop, db);
+  pendingTest.wptWaterfall = wptWaterfall;
+  pendingTest.wptFilmstrip = wptFilmstrip;
+
   return pendingTest
 }
 
+/**
+ * Creates the waterfall screenshot.
+ *
+ * @param wptTestId
+ * @param url
+ * @param isDesktop
+ * @param db- baqend database.
+ */
+async function createWaterfall(wptTestId: string, url: string, isDesktop: boolean = true, db:baqend): Promise<binding.File | null> {
+  try {
+    const device = isDesktop ? 'desktop' : 'mobile'
+    const screenshotLink = `https://wpt.baqend.com/waterfall.php?test=${wptTestId}`;
```

Comment:
Instead of writing the dns here you could also use the variable `credentials.wpt_dns`